src/main/java/pl/mmieczak/springbootexercise_22_1/UserRepository.java

```diff
@@ -22,4 +24,40 @@ public UserRepository() {
     public void add(String name, String surname, int age) {
         userCollection.add(new User(name, surname, age));
     }
+
+    public void removeAll(Stream<User> streamToRemove) {
+        userCollection.removeAll(streamToRemove.collect(Collectors.toList()));
+    }
+
+    public Stream<User> filterDataUsingFormValues(String name, String surname, String age){
+        Stream<User> userStream = userCollection.stream();
+        if (!name.equals("") && surname.equals("") && age.equals("-1")) {           //100
```

Comment:
Tutaj oczywiście ten sam komentarz co wcześniej apropo magicznej wartości "-1" która powinna być raczej sprawdzania pod kątem bycia `null`

Reply:
jakoś z poprzednich wykładów zapamiętałem ze porownywanie sie z nullem to nie jest dobra praktyka, ale masz racje ze z tym -1, to nawet jeszcze gorzej ;) zapamiętam....